DpcWaitOnProcess/Main.cpp

```diff
@@ -95,21 +86,22 @@ void CreateProcessNotifyRoutineEx (
     if (g_DpcWait == nullptr)
```

Comment:
There is a race condition here. You should use InterlockedCompareExchange when reading/setting this (and use a local) or have a boolean under a lock (or an interlock). Think of the "init once" pattern basically (you can even use those APIs if you want to).